lib/rsvp/enumerator.js

```diff
@@ -18,9 +17,8 @@ import ownResolve from './promise/resolve';
 export default class Enumerator {
   constructor(Constructor, input, abortOnReject, label) {
     this._instanceConstructor = Constructor;
-    this.promise = new Constructor(noop, label);
+    this.promise = new OwnPromise(noop, label);
```

Comment:
this should be always `OwnPromise` else `subscribe/publish` system won't work

Reply:
This doesn't seem right.

The following must remain true `Subclass.all([1]).then() instanceof Subclass`